tools/monorepo-utils/src/changelog/lib/__tests__/projects.ts

```diff
@@ -0,0 +1,140 @@
+			'interesting-project',
+		];
+
+		expectedProjects.forEach( ( expectedProject ) => {
+			expect( projects ).toContain( expectedProject );
+		} );
+
+		expect( projects ).toHaveLength( expectedProjects.length );
+	} );
+
+	it( 'getChangeloggerProjectPaths should provide a list of all projects that use Jetpack changelogger', async () => {
+		const projects = await getAllProjectsPathsFromWorkspace(
+			tmpRepoPath,
+			sampleWorkspaceYaml
+		);
+		const changeloggerProjects = await getChangeloggerProjectPaths(
+			tmpRepoPath,
+			projects
+		);
+
+		const expectedChangeLoggerProjects = [
+			'folder-with-lots-of-projects/project-b',
+			'folder-with-lots-of-projects/project-a',
+			'projects/very-cool-project',
+		];
+
+		expectedChangeLoggerProjects.forEach(
+			( expectedChangeLoggerProject ) => {
+				expect( projects ).toContain( expectedChangeLoggerProject );
+			}
+		);
+
+		expect( changeloggerProjects ).toHaveLength(
+			expectedChangeLoggerProjects.length
+		);
+	} );
+
+	it( 'getTouchedChangeloggerProjectsPathsMappedToProjects should combine touched and changelogger projects and return a list that is a subset of both', async () => {
+		const touchedFiles = [
+			'folder-with-lots-of-projects/project-b/src/index.js',
+			'projects/very-cool-project/src/index.js',
+		];
+		const changeLoggerProjects = [
+			'folder-with-lots-of-projects/project-b',
+			'folder-with-lots-of-projects/project-a',
+			'projects/very-cool-project',
+		];
+		const intersectedProjects =
+			getTouchedChangeloggerProjectsPathsMappedToProjects(
+				touchedFiles,
+				changeLoggerProjects
+			);
+
+		expect( intersectedProjects ).toHaveLength( 2 );
+		expect( intersectedProjects ).toContain(
+			'folder-with-lots-of-projects/project-b'
+		);
+		expect( intersectedProjects ).toContain( 'projects/very-cool-project' );
+	} );
+
+	it( 'getTouchedChangeloggerProjectsPathsMappedToProjects should map plugins and js packages to the correct name', async () => {
+		const touchedFiles = [
+			'plugins/beta-tester/src/index.js',
+			'plugins/woocommerce/src/index.js',
+			'packages/js/components/src/index.js',
+			'packages/js/data/src/index.js',
+		];
+		const changeLoggerProjects = [
+			'plugins/woocommerce',
+			'plugins/beta-tester',
+			'packages/js/data',
+			'packages/js/components',
+		];
+		const intersectedProjects =
+			getTouchedChangeloggerProjectsPathsMappedToProjects(
+				touchedFiles,
+				changeLoggerProjects
+			);
+
+		expect( intersectedProjects ).toHaveLength( 4 );
+		expect( intersectedProjects ).toContain( 'woocommerce' );
+		expect( intersectedProjects ).toContain( 'beta-tester' );
+		expect( intersectedProjects ).toContain( '@woocommerce/components' );
+		expect( intersectedProjects ).toContain( '@woocommerce/data' );
+	} );
+
+	it( 'getTouchedChangeloggerProjectsPathsMappedToProjects should handle woocommerce-admin projects mapped to woocommerce core', async () => {
+		const touchedFiles = [
+			'plugins/beta-tester/src/index.js',
```

Comment:
I feel like the inclusion of `beta-tester` distracts a bit from the purpose of this test and could be removed, especially since that's covered by the prior test.

Reply:
fixed